app/src/main/java/com/example/learn09082021v001/DataInput.kt

```diff
@@ -1,56 +1,72 @@
 package com.example.learn09082021v001
 
+import android.app.Activity
 import android.content.Intent
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.text.Editable
 import android.util.Log
-import android.widget.Toast
+import androidx.activity.result.ActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
 import kotlinx.android.synthetic.main.activity_data_input.*
 
 
 class DataInput : AppCompatActivity() {
+    override fun onSaveInstanceState(savedInstanceState: Bundle) {
+        super.onSaveInstanceState(savedInstanceState)
+        savedInstanceState.putString("nameandroid", edtNameAndroid.text.toString())
```

Comment:
create a constant class to handle all of these keys, don't use hard code
"nameandroid"
"scoreandroi"
"nameios"